pippy/IR.py

```diff
@@ -1178,34 +1179,31 @@ def _get_param_buffer_mapping(
         return param_buffer_table
 
 
-class PipeSplitWrapper(torch.nn.Module):
-    class SplitPoint(Enum):
-        BEGINNING = 1
-        END = 2
+class SplitPoint(Enum):
+    BEGINNING = 1
+    END = 2
 
-    def __init__(
-        self,
-        mod: torch.nn.Module,
-        split_point: SplitPoint = SplitPoint.BEGINNING,
-    ):
-        super().__init__()
-        self.mod = mod
-        self.split_point = split_point
 
-    def forward(self, *args, **kwargs):
-        try:
-            if self.split_point == self.SplitPoint.BEGINNING:
-                pipe_split()
+# For backward compatibility, we kept the PipeSplitWrapper class because `class
+# SplitPoint` used to be defined in this class.
+class PipeSplitWrapper:
+    # Create a class alias for BC
+    SplitPoint = SplitPoint
 
-            return self.mod(*args, **kwargs)
-        finally:
-            if self.split_point == self.SplitPoint.END:
-                pipe_split()
 
+def _split_before_forwad(self, *args, **kwargs):
```

Comment:
minor but helpful - can we update to proper spelling of 'forward' (currently is missing an r, 'forwad') esp since self._orig_forward right below is correct spelling. 
It can be confusing otherwise.  (might seem nitpicky but had an issue before that all boiled down to a common word mis-spelled in the api vs callers were spelling that word correctly).